api_tests/lib/bitcoind_instance.ts

```diff
@@ -18,39 +18,25 @@ export class BitcoindInstance implements LedgerInstance {
         private readonly logDir: string,
         public readonly p2pPort: number,
         public readonly rpcPort: number
-    ) {}
+    ) {
+        this.dbDir = tmp.dirSync();
+        this.writeLogFile();
```

Comment:
What's the reason behind doing this in the constructor? I think this is better done in the `start()` function.

Reply:
I don't have a strong opinion on it.  For lnd we don't pass the bitcoind data dir out of the constructor so that's why its done in the constructor.  Happy to re-consider if you have a technical reason why not to do it in the constructor otherwise I'd prefer to see the bitcoind and lnd code done in a uniform manner.

Reply:
Turns out that doing as you suggest resolves the async/sync constructor mix up pointed out over on #2020 by @thomas.  So implementing as suggested.